### PR TITLE
spirv-val: Fix Vulkan image sampled check

### PR DESCRIPTION
**Pull request by sfricke-samsung**
Fix to match the Vulkan `VUID-StandaloneSpirv-OpTypeImage-04656`

> OpTypeImage must declare a scalar 32-bit float, 64-bit integer, or 32-bit integer type for the “Sampled Type” (RelaxedPrecision can be applied to a sampling instruction and to the variable holding the result of a sampling instruction)

Added more test coverage of various types as well

### CODE REVIEW

File: source/val/validate_image.cpp [hunk 1]
```diff
@@ -750,12 +750,14 @@ spv_result_t ValidateTypeImage(ValidationState_t& _, const Instruction* inst) {
   if (spvIsVulkanEnv(target_env)) {
     if ((!_.IsFloatScalarType(info.sampled_type) &&
          !_.IsIntScalarType(info.sampled_type)) ||
-        (32 != _.GetBitWidth(info.sampled_type) &&
-         (64 != _.GetBitWidth(info.sampled_type) ||
-          !_.HasCapability(SpvCapabilityInt64ImageEXT)))) {
-      return _.diag(SPV_ERROR_INVALID_DATA, inst)
-             << "Expected Sampled Type to be a 32-bit int or float "
-                "scalar type for Vulkan environment";
+        ((32 != _.GetBitWidth(info.sampled_type)) &&
```
Comment (alan-baker) [inline source/val/validate_image.cpp:753]: Why did you change the logic? It looked correct previously, just the error message seemed out of date.

Comment (sfricke-samsung) [reply source/val/validate_image.cpp:753]: hmm... I "ment" to change the logic, I realize what I am missing is a test with both a `Float64` and `Int64` as it would have a `Int64ImageEXT` capability and make it through

Comment (sfricke-samsung) [reply source/val/validate_image.cpp:761]: should be fixed in updated commit now, move capability check out to Universal scope

**Review by alan-baker — APPROVED** [review]
